sources/models/src/modeled_types/ecs.rs

```diff
@@ -228,3 +228,57 @@ mod test_ecs_agent_log_level {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// ECSAgentImagePullBehavior represents a string that contains a valid ECS Image Pull Behavior for the ECS agent.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct ECSAgentImagePullBehavior {
+    inner: String,
+    pub inner_code: String,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+enum ECSImagePullBehavior {
+    Default = 0,
+    Always,
+    Once,
+    PreferCached,
+}
+
+string_impls_for!(ECSAgentImagePullBehavior, "ECSAgentImagePullBehavior");
+
+impl TryFrom<&str> for ECSAgentImagePullBehavior {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        serde_plain::from_str::<ECSImagePullBehavior>(&input).context(error::InvalidPlainValueSnafu {
+            field: "ecs.image_pull_behavior",
+        })?;
+        Ok(ECSAgentImagePullBehavior {
+            inner: input.to_string(),
+            inner_code: (serde_plain::from_str::<ECSImagePullBehavior>(&input).unwrap() as u8).to_string(),
+        })
```

Comment:
This avoids the `unwrap()` which nominally can panic, and the second conversion attempt:
```suggestion
        let behavior = serde_plain::from_str::<ECSImagePullBehavior>(&input).context(error::InvalidPlainValueSnafu {
            field: "ecs.image_pull_behavior",
        })?;
        Ok(ECSAgentImagePullBehavior {
            inner: input.to_string(),
            inner_code: (behavior as u8).to_string(),
        })
```